rust-src/crypto_common_derive/src/lib.rs

```diff
@@ -25,13 +25,13 @@ pub fn serde_base16_serialize_derive(input: TokenStream) -> TokenStream {
     let ident_deserializer = format_ident!("deserializer", span = span);
     let gen = quote! {
         impl #impl_generics SerdeSerialize for #name #ty_generics #where_clauses {
-            fn serialize<#ident: serde::Serializer>(&self, #ident_serializer: #ident) -> Result<#ident::Ok, #ident::Error> {
+            fn serialize<#ident: serde::Serializer>(&self, #ident_serializer: #ident) -> ParseResult<#ident::Ok, #ident::Error> {
```

Comment:
This is a Serde trait implementation, so it should use `std::Result` not `ParseResult`.